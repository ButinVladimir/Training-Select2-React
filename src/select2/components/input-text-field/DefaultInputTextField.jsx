import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

export default function DefaultInputTextField({
  searchQuery,
  onChange,
}) {
  return (
    <input className="select2-input-text-field" type="text" value={searchQuery} onChange={onChange} />
  );
}

DefaultInputTextField.propTypes = {
  searchQuery: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
