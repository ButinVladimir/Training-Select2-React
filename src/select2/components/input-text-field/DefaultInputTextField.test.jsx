import React from 'react';
import renderer from 'react-test-renderer';
import DefaultInputTextField from './DefaultInputTextField';

describe('DefaultInputTextField', () => {
  test('renders without crashing', () => {
    const mockFn = jest.fn();
    const component = renderer.create(<DefaultInputTextField searchQuery="123" onChange={mockFn} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockFn).not.toHaveBeenCalled();
  });

  test('fires onChange properly', () => {
    const mockFn = jest.fn();
    const component = renderer.create(<DefaultInputTextField searchQuery="123" onChange={mockFn} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onChange();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockFn).toHaveBeenCalled();
  });
});
