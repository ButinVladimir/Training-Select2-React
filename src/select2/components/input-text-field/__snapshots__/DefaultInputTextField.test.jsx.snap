// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultInputTextField fires onChange properly 1`] = `
<input
  className="select2-input-text-field"
  onChange={[Function]}
  type="text"
  value="123"
/>
`;

exports[`DefaultInputTextField fires onChange properly 2`] = `
<input
  className="select2-input-text-field"
  onChange={[Function]}
  type="text"
  value="123"
/>
`;

exports[`DefaultInputTextField renders without crashing 1`] = `
<input
  className="select2-input-text-field"
  onChange={[Function]}
  type="text"
  value="123"
/>
`;
